app/controllers/admin/workshops_controller.rb

```diff
@@ -122,12 +122,11 @@ def workshop_id
 
   def set_host(host_id)
     return unless host_id
-
     host = @workshop.workshop_sponsors.find_or_initialize_by(sponsor_id: host_id)
-    unless @workshop.host.eql?(host.sponsor)
-      @workshop.workshop_sponsors.where(sponsor: @workshop.host).destroy_all
-      host.update(host: true)
-    end
+    return if @workshop.host.eql?(host.sponsor)
```

Comment:
@matyikriszta - If you change this from `if` to `unless` it breaks the code. You can change any of these guard clauses in the commit from `unless` to `if` or `if` to `unless` and tests break. I didn't check in a couple of the refactorings because they weren't backed by tests and I won't do a change until it is.